pkg/operator/ceph/pool/radosnamespace/controller.go

```diff
@@ -261,6 +262,45 @@ func (r *ReconcileCephBlockPoolRadosNamespace) reconcile(request reconcile.Reque
 		return reconcile.Result{}, errors.Wrap(err, "failed to save cluster config")
 	}
 
+	poolAndRadosNamespaceName := fmt.Sprintf("%s/%s", pool, getRadosNamespaceName(cephBlockPoolRadosNamespace))
+	if cephBlockPoolRadosNamespace.Spec.Mirroring.Enabled {
+		err = cephclient.EnableRBDRadosNamespaceMirroring(r.context, r.clusterInfo, poolAndRadosNamespaceName, cephBlockPoolRadosNamespace.Spec.Mirroring.RemoteNamespace, string(cephBlockPoolRadosNamespace.Spec.Mirroring.Mode))
+		if err != nil {
+			return reconcile.Result{}, errors.Wrap(err, "failed to enable rbd rados namespace mirroring")
+		}
+
+		// Schedule snapshots
+		if len(cephBlockPoolRadosNamespace.Spec.Mirroring.SnapshotSchedules) > 0 {
+			err = cephclient.EnableSnapshotSchedules(r.context, r.clusterInfo, poolAndRadosNamespaceName, cephBlockPoolRadosNamespace.Spec.Mirroring.SnapshotSchedules)
+			if err != nil {
+				return reconcile.Result{}, errors.Wrapf(err, "failed to enable snapshot scheduling for rbd rados namespace %q", poolAndRadosNamespaceName)
+			}
+		}
+	} else {
+		mirrorInfo, err := cephclient.GetPoolMirroringInfo(r.context, r.clusterInfo, poolAndRadosNamespaceName)
```

Comment:
Does this mirrorInfo indicate if mirroring is actually enabled on the rados namespace? If it's not enabled, then no need to check for the images below, or call Disable again.

Reply:
Yaa it wil be `disabled` adding a check

Reply:
We can also improve the pool recocnile with this

Reply:
Sounds great to have that check. I assume you're still working on it, I don't see it.

Reply:
I have already added it,
1) We get output `mirrorInfo,` var
2) Then we check if `mirrorInfo.mode==image`
3) Then we check if `mirrorInfo.mode!=disabled`


I try to keep it separate instead of a big `if and else`.